bug: cannot convert byte to int with int()

## Description
The `int()` type conversion function does not support converting `byte` values to `int`, even though the reverse conversion (`byte()` from int) is supported.

## Expected Behavior
`int(byte_value)` should return the integer representation of the byte (0-255).

## Reproduction
```ez
import @std
using std

do main() {
    temp by byte = byte(255)
    temp by_int int = int(by)  // Error: cannot convert BYTE to int
    println("Result: ${by_int}")
}
```

## Actual Behavior
```
error[E7014]: cannot convert BYTE to int
  --> test.ez:6:26
   |
6  |     temp by_int int = int(by)
   |                          ^ error occurred here
```

## Why This Is A Bug
- Asymmetric conversion: `byte(int_val)` works but `int(byte_val)` doesn't
- The `char` type supports `int(char_val)` to get ASCII code
- Common use case: reading binary data and converting to integers for manipulation
- No workaround exists for getting the integer value of a byte

--- a/pkg/stdlib/builtins.go
+++ b/pkg/stdlib/builtins.go
@@ -250,6 +250,8 @@ var StdBuiltins = map[string]*object.Builtin{
 				return &object.Integer{Value: val}
 			case *object.Char:
 				return &object.Integer{Value: int64(arg.Value)}
+			case *object.Byte:
+				return &object.Integer{Value: int64(arg.Value)}
 			case *object.EnumValue:
 				// Extract the underlying value from the enum
 				switch v := arg.Value.(type) {

